feat(crons): Add product docs for environments (#6679)
getsentry/sentry-docs: src/docs/product/crons/job-monitoring.mdx

The table of recent check-ins lists previously scheduled jobs and their status.

> ## Multiple environments
> 
> 
> It is possible to send check-ins for multiple environments to one monitor with
> a shared schedule. This may be useful if you wish to monitor the same job in
> different environments. Each environment has their own status and set of
> check-ins.

Reviewer: ```suggestion
## Multiple Environments

To monitor the same job in different environments, use a shared schedule to send check-ins for multiple environments to one monitor. Each environment will have its own status and set of check-ins.
```

> ### Alerting on specific environments
> 
> If you wish to only receive alerts for failing / missed monitor alerts for a
> specific environment or set of environments, you can simply [configure the
> environment](/product/alerts/create-alerts/issue-alert-config/#environment) for
> your monitor alert.

Reviewer: ```suggestion
### Alerting on Specific Environments

To only receive failing or missed monitor alerts for a specific environment or set of environments, [configure the
environment](/product/alerts/create-alerts/issue-alert-config/#environment) for
your monitor alert.
```

> ### Existing limitations:
> 
> - It is not currently possible to pause an individual monitor environment.
> 
> - It is not currently possible to see an aggregate view of a monitors status
>   across all environments, each environment will appear as it's own row in the
>   monitor list.
> 
> - Currently monitors environments cannot be deleted, only entire monitors may
>   be removed.

Reviewer: ```suggestion
### Existing Limitations:

- It's not currently possible to pause an individual monitor environment.

- It's not currently possible to see an aggregate view of a monitors' status
  across all environments. Each environment will appear as its own row in the
  monitor list.

- Currently, only entire monitors can be removed. Individual monitor environments can't be deleted. 
```

The table of recent check-ins lists previously scheduled jobs and their status.

## Multiple environments

> <Alert level="info">
>     Environment support is still in it's early stages, the user experience may
>     have some kinks we've yet to work out. Feel free to email
>     [crons-feedback@sentry.io](mailto:crons-feedback@sentry.io) with any
>     feedback!
> </Alert>

Reviewer 2: Do you think we really need this? Since in our main Crons page and set up page we already have a warning that Crons by itself is in beta.

```suggestion
```

Author: Yeah we can drop it I think you're right
